### PR TITLE
Support Changing Number of Tensor Parallel Partitions for Megatron

### PR DESCRIPTION
**Pull request by aklife97**
Added `megatron_change_num_partitions.py` that supports converting a Megatron model with an arbitrary number of Tensor Parallel (TP) partitions N to any other number of TP partitions M.

In order to do so, the script intermediately converts all models to TP=1.
All the models/tensors are created/loaded on CPU.

Usage:
```
python examples/nlp/language_modeling/megatron_change_num_partitions.py \
   --model_file=PATH_TO_SRC_FILE \
   --target_file=PATH_TO_TGT_FILE \
   --tensor_model_parallel_size=2 \
   --target_tensor_model_parallel_size=1
```

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1962,6 +1962,28 @@ pipeline {
...
+      steps{
+        sh "python examples/nlp/language_modeling/megatron_change_num_partitions.py \
+            --model_file \
+            /home/TestData/nlp/megatron_gpt/125M/megatron_gpt.nemo \
+            --target_file \
+            /home/TestData/nlp/megatron_gpt/125M/test-split.nemo \
+            --tensor_model_parallel_size \
+            1 \
+            --target_tensor_model_parallel_size \
+            2"
```
Comment (ericharper) [inline Jenkinsfile:1982]: Small nitpick, but the case we're most interested in is from larger num partitions -> smaller num partitions. I'd rather we test that case in our CI.

File: examples/nlp/language_modeling/megatron_change_num_partitions.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+
+"""
+Usage:
+python megatron_change_num_partitions.py \
+    --model_file=PATH_TO_SRC_FILE \
+    --target_file=PATH_TO_TGT_FILE \
+    --tensor_model_parallel_size=2 \
+    --target_tensor_model_parallel_size=1
+"""
```
Comment (ericharper) [inline examples/nlp/language_modeling/megatron_change_num_partitions.py:33]: Could you also add here what models are supported?

File: examples/nlp/language_modeling/megatron_change_num_partitions.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    parser = ArgumentParser()
+    parser.add_argument("--model_file", type=str, required=True, help="Path to source .nemo file")
+    parser.add_argument("--target_file", type=str, required=True, help="Path to write target .nemo file")
+    parser.add_argument("--tensor_model_parallel_size", type=int, required=True, help="TP size of source model")
+    parser.add_argument("--target_tensor_model_parallel_size", type=int, required=True, help="TP size of target model")
+    parser.add_argument(
+        "--model_class",
+        type=str,
+        default="nemo.collections.nlp.models.language_modeling.megatron_gpt_model.MegatronGPTModel",
+        help="NeMo model class",
```
Comment (ericharper) [inline examples/nlp/language_modeling/megatron_change_num_partitions.py:131]: Can add supported models here.

File: examples/nlp/language_modeling/megatron_change_num_partitions.py [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+Usage:
+python megatron_change_num_partitions.py \
+    --model_file=PATH_TO_SRC_FILE \
+    --target_file=PATH_TO_TGT_FILE \
+    --tensor_model_parallel_size=2 \
+    --target_tensor_model_parallel_size=1
+"""
+
+
+def merge_parition(model, partitions, write_path=None):
```
Comment (ericharper) [inline examples/nlp/language_modeling/megatron_change_num_partitions.py:36]: typo: this should be `merge_partition`

**Review by ericharper — CHANGES_REQUESTED** [review]
Overall looks good, just some minor comments and typos. 
Can you add some documentation on what models are supported? And also a programmatic check?

File: Jenkinsfile (hunk 1 above)
Comment (aklife97) [reply Jenkinsfile]: Changed CI to 2->1

File: examples/nlp/language_modeling/megatron_change_num_partitions.py (hunk 2 above)
Comment (aklife97) [reply examples/nlp/language_modeling/megatron_change_num_partitions.py:33]: Added details in the argparse help

File: examples/nlp/language_modeling/megatron_change_num_partitions.py (hunk 3 above)
Comment (aklife97) [reply examples/nlp/language_modeling/megatron_change_num_partitions.py]: Added details in the help

File: examples/nlp/language_modeling/megatron_change_num_partitions.py (hunk 4 above)
Comment (aklife97) [reply examples/nlp/language_modeling/megatron_change_num_partitions.py]: Fixed!

**Review by ericharper — APPROVED** [review]
LGTM. Thanks for the changes!

**Pull request merged**